packages/framework/src/client/scheduler.js

```diff
@@ -79,91 +90,141 @@ export default class Scheduler {
+			);
 		} );
 	};
 
 	/**
 	 * Finds requests that are in flight by resourceName
 	 * @param {string} resourceName The name of the resource to check
-	 * @param {Date} now The current time
+	 * @param {string} operation The name of the operation to be performed (defaults to 'read')
+	 * @param {Date} now The current time.
 	 * @return {Array} Any requests for the given resource which are currently in flight.
 	 */
-	getInFlightRequests = ( resourceName, now = new Date() ) => {
+	getInFlightRequests = ( resourceName, operation, now = new Date() ) => {
 		return this.requests.filter( ( r ) => {
-			return ( resourceName === r.resourceName && STATUS.inFlight === r.getStatus( now ) );
+			return (
+				resourceName === r.resourceName &&
+				operation === r.operation &&
+				STATUS.inFlight === r.getStatus( now )
+			);
 		} );
 	};
 
 	/**
-	 * Schedule a read of a resource name according to the requirement set given.
-	 * @param {string} resourceName The name of the resource to be read
+	 * Schedule a operation on a resource name according to the requirement set given.
 	 * @param {Object} requirement The set of requirements (freshness, timeout)
 	 * @param {Object} resourceState The current snapshot of resource state
+	 * @param {string} resourceName The name of the resource for the operation
+	 * @param {string} operation The name of the operation to be performed (defaults to 'read')
+	 * @param {Object} data The data to be sent for the operation (defaults to undefined)
 	 * @param {Date} now The current time.
 	 */
-	scheduleRequest = ( resourceName, requirement, resourceState, now = new Date() ) => {
-		if ( this.getInFlightRequests( resourceName, now ).length > 0 ) {
-			// Do nothing, there's already a request in flight.
+	scheduleRequest = (
+		requirement,
+		resourceState,
+		resourceName,
+		operation,
+		data = undefined,
+		now = new Date()
+	) => {
+		const identicalInFlightRequest = find(
+			this.getInFlightRequests( resourceName, operation, now ),
+			( request ) => {
+				return isEqual( request.data, data ) || ( isNil( request.data ) && isNil( data ) );
+			}
+		);
+
+		if ( identicalInFlightRequest ) {
+			// Do nothing, there's already an identical request in flight.
 			return;
 		}
 
-		const existingRequest = this.getScheduledRequest( resourceName, now );
+		const existingRequest = this.getScheduledRequest( resourceName, operation, now );
 		if ( existingRequest ) {
-			existingRequest.addRequirement( requirement, resourceState, now );
+			existingRequest.append( requirement, resourceState, data, now );
 		} else {
-			this.requests.push( new ResourceRequest( resourceName, requirement, resourceState, now ) );
+			this.requests.push( new ResourceRequest( requirement, resourceState, resourceName, operation, data, now ) );
 		}
 		this.updateDelay( now );
 	};
 
+	// TODO: Remove this after mutations are updated to no longer use operations directly.
+	scheduleMutationOperation = (
```

Comment:
I'm not sure I understand `scheduleMutationOperation` and `scheduleRequest` - looks like they're used in later PRs though so I'll wait until then.

Reply:
`scheduleRequest` is the important one. In fact, `scheduleMutationOperation` calls it, and is only there to marshall backward compatibility with the current way mutations work with the api spec.